### PR TITLE
feat(contexts): Normalize known os.name to render UI context icons`

### PR DESCRIPTION
**Pull request by bruno-garcia**
Following efforts to [send more contextual data from the C# SDK](https://github.com/getsentry/raven-csharp/pull/237), this PR intends to improve the visualization of the data extracted from the clients.

### CODE REVIEW

**Review by mitsuhiko — CHANGES_REQUESTED** [review]
I really think we should map from `raw_description` to `name` etc. instead of reusing `name` for this. I see value in the original data.

**Comment by bruno-garcia** [comment]
@mitsuhiko in order to do that I've sent a PR on the SDK first to send the data via `raw-description`:  getsentry/raven-csharp/pull/243

I'll change the parsing to consider this new fields instead.

File: src/sentry/interfaces/contexts.py [hunk 1]
```diff
@@ -138,10 +139,15 @@ class OsContextType(ContextType):
     indexed_fields = {
         '': u'{name} {version}',
         'name': u'{name}',
+        'raw_description': u'{raw_description}',
```
Comment (jan-auer) [inline src/sentry/interfaces/contexts.py]: I think we should not create tags for `raw_description`

File: src/sentry/utils/os_normalization.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Environment.OSVersion (GetVersionEx) or RuntimeInformation.OSDescription, on Windows
+windows = re.compile('^(Microsoft )?Windows (NT )?(?P<version>\d+\.\d+\.\d+).*$')
+# Environment.OSVersion or RuntimeInformation.OSDescription (uname)
+# on Mono and CoreCLR on macOS, iOS, Linux, etc
+uname = re.compile('^(?P<name>[a-zA-Z]+) (?P<version>\d+\.\d+\.\d+(\.[1-9]+)?).*$')
+
+
+# If there's no name and version, attempts to infer from raw_description
+def normalize(data):
+    raw_description = data.get('raw_description', None)
```
Comment (jan-auer) [inline src/sentry/utils/os_normalization.py]: _nit:_ `data.get('foo')` will default to `None`, so you could omit the second parameter.

**Review by jan-auer — APPROVED** [review]
LGTM in general, just note the one comment on `indexed_fields`.

File: src/sentry/utils/os_normalization.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import
+import re
+
+# Environment.OSVersion (GetVersionEx) or RuntimeInformation.OSDescription, on Windows
+windows = re.compile('^(Microsoft )?Windows (NT )?(?P<version>\d+\.\d+\.\d+).*$')
+# Environment.OSVersion or RuntimeInformation.OSDescription (uname)
+# on Mono and CoreCLR on macOS, iOS, Linux, etc
+uname = re.compile('^(?P<name>[a-zA-Z]+) (?P<version>\d+\.\d+\.\d+(\.[1-9]+)?).*$')
```
Comment (HazAT) [inline src/sentry/utils/os_normalization.py]: nit: Most regexes in sentry have this naming style `_windows_re`, `_uname_re`

File: src/sentry/utils/os_normalization.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+
+# If there's no name and version, attempts to infer from raw_description
```
Comment (HazAT) [inline src/sentry/utils/os_normalization.py]: nit: Move this comment into the function above the line.

**Review by HazAT — CHANGES_REQUESTED** [review]

File: src/sentry/utils/os_normalization.py (hunk 3 above)
Comment (mitsuhiko) [reply src/sentry/utils/os_normalization.py]: I would add `(?i)` to make it case insensitive (the windows one)

**Review by mitsuhiko — APPROVED** [review]

**Review by HazAT — APPROVED** [review]

**Pull request merged**